node/test/go/dev-mode/test/integration/index.test.js

```diff
@@ -0,0 +1,608 @@
+                  let state;
+                  do {
+                    await wait(100);
+                    ({
+                      Configuration: { State: state },
+                    } = await awsRequest(Lambda, 'getFunction', {
+                      FunctionName: urlEndpointLambdaName,
+                    }));
+                  } while (state !== 'Active');
+                },
+                beforeDelete: async (testConfig) => {
+                  await Promise.all([
+                    awsRequest(Lambda, 'deleteFunctionUrlConfig', {
+                      FunctionName: testConfig.urlEndpointLambdaName,
+                      Qualifier: 'url',
+                    }),
+                    awsRequest(Lambda, 'deleteFunction', {
+                      FunctionName: testConfig.urlEndpointLambdaName,
+                    }),
+                  ]);
+                },
+              },
+              invokePayload: (testConfig) => {
+                return { url: `${testConfig.functionUrl}?foo=bar` };
+              },
+            },
+          ],
+        ]),
+      },
+    ],
+  ]);
+
+  const testVariantsConfig = resolveTestVariantsConfig(useCasesConfig);
+
+  before(async () => {
+    await createCoreResources(coreConfig);
+    for (const testConfig of testVariantsConfig) {
+      testConfig.deferredResult = processFunction(testConfig, coreConfig).catch((error) => ({
+        // As we process result promises sequentially step by step in next turn, allowing them to
+        // reject will generate unhandled rejection.
+        // Therefore this scenario is converted to successuful { error } resolution
+        error,
+      }));
+    }
+  });
+
+  for (const testConfig of testVariantsConfig) {
+    it(testConfig.name, async () => {
+      const testResult = await testConfig.deferredResult;
+      if (testResult.error) throw testResult.error;
+      log.debug('%s test result: %o', testConfig.name, testResult);
+      const { expectedOutcome } = testConfig;
+      const { invocationsData } = testResult;
+      if (expectedOutcome === 'success' || expectedOutcome === 'error:handled') {
+        if (expectedOutcome === 'success' && !testConfig.isAsyncInvocation) {
+          for (const { responsePayload } of invocationsData) {
+            expect(responsePayload.raw).to.equal('"ok"');
+          }
+        }
+        for (const [, logs] of invocationsData.map((data) => data.logs).entries()) {
+          const logPayload = logs[0];
+          const data = (logPayload.messages || [])[0] || { message: '' };
+          expect(logs.length).to.equal(1);
+          expect(logPayload.name).to.equal(testConfig.configuration.FunctionName);
+          expect(testConfig.name).to.have.string(
+            data.message.slice(data.message.lastIndexOf('\t') + 1).replace('\n', '')
+          );
+        }
+      }
+      if (testConfig.test) {
+        testConfig.test({ invocationsData, testConfig });
+      }
+    });
+  }
```

Comment:
This tests seems to test internal extension, while this is already well tested in its context.

Scope of this should be purely to confirm that external extension does it's job.

Same as in otel case, it'll be probably good to introduce mode where external extension payloads are written to CW logs and not send to server, and here were should read this logs and confirm they're as expected

Reply:
Yeah this should be gone with the slimmed down testing set 👍